src/main/java/de/muenchen/isi/infrastructure/repository/stammdaten/UmlegungFoerderartenRepository.java

```diff
@@ -1,10 +1,14 @@
 package de.muenchen.isi.infrastructure.repository.stammdaten;
 
 import de.muenchen.isi.infrastructure.entity.stammdaten.UmlegungFoerderarten;
+import java.time.LocalDate;
 import java.util.Optional;
 import java.util.UUID;
 import org.springframework.data.jpa.repository.JpaRepository;
 
 public interface UmlegungFoerderartenRepository extends JpaRepository<UmlegungFoerderarten, UUID> {
-    Optional<UmlegungFoerderarten> findBybezeichnung(final String bezeichnung);
+    Optional<UmlegungFoerderarten> findByBezeichnungAndGueltigAbBeforeOrderByGueltigAbDesc(
```

Comment:
Anstelle `findBy...` besser `findFirstBy...` verwenden. Da hier bereits im SQL-Statement die Ergebnismenge auf ein Element reduziert wird. Somit wird die Datenextraktion beschleunigt, da in der Datenbank due Datenextraktion nach dem ersten Element endet.

[findFirst](https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.limit-query-result)

Reply:
Das SQL-Statement wird dann ungefähr so aussehen.

```sql

SELECT y,x,z,... FROM Testinger
WHERE xxxx
FETCH FIRST 1 ROWS ONLY;

```